programs/Makefile

```diff
@@ -67,7 +67,10 @@ endif
 
 # zlib detection
 VOID = /dev/null
-HAVE_ZLIB := $(shell echo "\#include <zlib.h>\nint main(){}" | $(CC) -o $(VOID) -x c - -lz 2> $(VOID) && echo 1 || echo 0)
+HAVE_ZLIB := $(shell echo -e "\#include <zlib.h>\nint main(){}" | $(CC) -o have_zlib -x c - -lz 2> $(VOID) && echo 1 || echo 0)
+ifeq ($(HAVE_ZLIB), 1)
```

Comment:
I'm pretty sure the second `ifeq ($(HAVE_ZLIB), 1)` is played like the first, so there seems to be no reason to separate them.
Only the 3rd one is triggered from within `zstd-internal`.

It's not possible to define or update variables from within the 3rd one.
It's not possible to start commands from the 1st and 2nd one
(but you can define and evaluate variable, hence `TMP := $(shell echo ...` would work)